src/views/CashlinkCreate.vue

```diff
@@ -0,0 +1,557 @@
+        }
+    }
+
+    private async initNetwork() {
+        const network = NetworkClient.Instance;
+        await network.init();
+        this.balanceUpdated = this.balanceUpdated || new Promise((resolve) => {
+            const wallets = this.wallets.slice(0);
+            let addresses: string[] = [];
+            let accountsAndContracts = Array<AccountInfo | ContractInfo>();
+            if (!this.accountOrContractInfo) { // No senderAccount in the request
+                accountsAndContracts = wallets.reduce((acc, wallet) => {
+                    acc.push(...wallet.accounts.values());
+                    acc.push(...wallet.contracts);
+                    return acc;
+                }, [] as Array<AccountInfo | ContractInfo>);
+                // Reduce userfriendly addresses from that
+                addresses = accountsAndContracts.map((accountOrContract) => accountOrContract.userFriendlyAddress);
+            } else {
+                accountsAndContracts.push(this.accountOrContractInfo);
+                addresses.push(this.accountOrContractInfo.userFriendlyAddress);
+            }
+            network.getBalance(addresses).then(async (balances) => {
+                for (const accountOrContract of accountsAndContracts) {
+                    const balance = balances.get(accountOrContract.userFriendlyAddress);
+                    if (balance === undefined) continue;
+
+                    if ('type' in accountOrContract && accountOrContract.type === Nimiq.Account.Type.VESTING) {
+                        // Calculate available amount for vesting contract
+                        accountOrContract.balance =
+                            (accountOrContract as VestingContractInfo).calculateAvailableAmount(
+                                NetworkClient.Instance.headInfo.height,
+                                Nimiq.Policy.coinsToSatoshis(balance),
+                            );
+                    } else {
+                        accountOrContract.balance = Nimiq.Policy.coinsToSatoshis(balance);
+                    }
+                }
+                // Store updated wallets
+                for (const wallet of wallets) {
+                    // Update IndexedDB
+                    await WalletStore.Instance.put(wallet);
+
+                    // Update Vuex
+                    this.$addWallet(wallet);
+                }
+                resolve();
+            });
+        });
+    }
+
+    private updateFeePreview(fee: number) {
+        this.feeLunaPerBytePreview = fee;
+    }
+
+    private setFee() {
+        this.optionsOpened = false;
+        this.feeLunaPerByte = (this.$refs.fee as SelectBar).value;
+        this.liveAmountAndFee.fee = this.fee;
+        Vue.nextTick(() => (this.$refs.amountWithFee as AmountWithFee).focus());
+    }
+
+    private get fee(): number {
+        return 171 * this.feeLunaPerByte; // 166 + 5 bytes extraData for funding a cashlink
```

Comment:
They are the same size. Tx from a vesting contract are also extended tx, just like this one (because we require the tx message anyway). Otherwise vesting payouts are regular tx, just signed by the owner of the contract.